extensions/pkg/bastion/vmdetails.go

```diff
@@ -0,0 +1,217 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package bastion
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/Masterminds/semver/v3"
+	"golang.org/x/exp/maps"
+	"k8s.io/utils/ptr"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+)
+
+// VMDetails define all bastion vm details derived from the CloudProfile
+type VMDetails struct {
+	MachineName   string
+	Architecture  string
+	ImageBaseName string
+	ImageVersion  string
+}
+
+// DetermineVmDetails determines the bastion vm details based on information in the cloud profile
+func DetermineVmDetails(spec gardencorev1beta1.CloudProfileSpec) (vm VMDetails, err error) {
+	imageArchs, err := getArchitectures(spec.Bastion, spec.MachineImages)
+	if err != nil {
+		return VMDetails{}, err
+	}
+	vm.MachineName, vm.Architecture, err = getMachine(spec.Bastion, spec.MachineTypes, imageArchs)
+	if err != nil {
+		return VMDetails{}, err
+	}
+	vm.ImageBaseName, err = getImageName(spec.Bastion, spec.MachineImages, vm.Architecture)
+	if err != nil {
+		return VMDetails{}, err
+	}
+	vm.ImageVersion, err = getImageVersion(vm.ImageBaseName, vm.Architecture, spec.Bastion, spec.MachineImages)
+	return vm, err
+}
+
+// getMachine retrieves the bastion machine name and arch
+func getMachine(bastion *gardencorev1beta1.Bastion, machineTypes []gardencorev1beta1.MachineType, supportedArchs []string) (machineName string, machineArch string, err error) {
+	if bastion != nil && bastion.MachineType != nil {
+		machineIndex := slices.IndexFunc(machineTypes, func(machine gardencorev1beta1.MachineType) bool {
+			return machine.Name == bastion.MachineType.Name
+		})
+
+		if machineIndex == -1 {
+			return "", "",
+				fmt.Errorf("bastion machine with name %s not found in cloudProfile", bastion.MachineType.Name)
+		}
+
+		machine := machineTypes[machineIndex]
+		return machine.Name, *machine.Architecture, nil
+	}
+
+	// find the machine in cloud profile with the lowest amount of cpus
+	var minCpu *int64
+
+	for _, machine := range machineTypes {
+		if machine.Architecture == nil {
```

Comment:
I think you can default to amd64 here

Reply:
Feels kind of wrong to use a default here in my opinion - personally I prefer to fail with an error.
Checked the cloud profile of all our providers - the architecture field should normally be set.